server/src/main/java/io/crate/types/StringType.java

```diff
@@ -123,39 +123,37 @@ public String value(Object value) {
         if (value == null) {
             return null;
         }
+        final String str;
         if (value instanceof String) {
-            return (String) value;
-        }
-        if (value instanceof BytesRef) {
-            return ((BytesRef) value).utf8ToString();
-        }
-        if (value instanceof Boolean) {
-            if ((boolean) value) {
-                return T;
-            } else {
-                return F;
-            }
-        }
-        if (value instanceof Map) {
+            str = (String) value;
+        } else if (value instanceof BytesRef) {
+            str = ((BytesRef) value).utf8ToString();
+        } else if (value instanceof Boolean) {
+            str = (boolean) value ? T : F;
+        } else if (value instanceof Map) {
             try {
                 //noinspection unchecked
-                return Strings.toString(XContentFactory.jsonBuilder().map((Map<String, ?>) value));
+                str = Strings.toString(XContentFactory.jsonBuilder().map((Map<String, ?>) value));
             } catch (IOException e) {
                 throw new IllegalArgumentException("Cannot cast `" + value + "` to type TEXT", e);
             }
-        }
-        if (value instanceof Collection) {
+        } else if (value instanceof Collection) {
             throw new IllegalArgumentException(
                 String.format(Locale.ENGLISH, "Cannot cast %s to type TEXT", value));
-        }
-        if (value.getClass().isArray()) {
+        } else if (value.getClass().isArray()) {
             throw new IllegalArgumentException(
                 String.format(Locale.ENGLISH, "Cannot cast %s to type TEXT", Arrays.toString((Object[]) value)));
+        } else if (value instanceof TimeValue) {
+            str = ((TimeValue) value).getStringRep();
+        } else {
+            str = value.toString();
         }
-        if (value instanceof TimeValue) {
-            return ((TimeValue) value).getStringRep();
+
+        if (unbound() || str.length() <= lengthLimit) {
```

Comment:
maybe we can optimize by creating two a dedicated unbound string type.

Reply:
Not sure if it is worth it. Maybe do a quick JMH benchmark to get a feeling of the impact?

Reply:
The benchmark shows no difference.
```
Benchmark                                          Mode  Cnt   Score    Error  Units
StringTypeTest.withoueUnboundOrLengthLimitCheck    avgt   25  ≈ 10⁻⁵           ms/op
StringTypeTest.withUnboundOrLengthLimitCheck       avgt   25  ≈ 10⁻⁵           ms/op
```